### DISCUSSION TITLE
Issue with Child Prefix since upgrading to 3.0.0

### DISCUSSION THREAD

**Discussion by aquelf** [netbox-community/netbox#7146] [Q&A]
I have recently upgraded from 2.11.7 to 3.0.0.

What I have noticed that under Sites > Prefixes > Child Prefixs I get the following Server Error?

![child prefixes](https://user-images.githubusercontent.com/84839725/131830335-71d623d7-e716-4c8a-8ec6-84d2b9a6faf0.PNG)

This was how I nested the subnets prior to upgrade, has something changed or broken after the upgrade, thanks?

**Comment by Bloodiko** [comment]
pls see Issue #7072 for Reference. 

This was a bug in 3.0.0 - and was fixed in yesterdays Release 3.0.1 

Bloodiko

**Reply by Bloodiko** [reply]
Oh , I'm sorry - its a different issue. 
I didnt look into the Screenshot. 

You are already on v3.0.1. 
And the Error is a different one. 

But my Sites doesnt have an Prefix. 
Do you have a custom Field added for it ?

**Reply by aquelf** [reply]
No so basically, to try make it more readable per site, On the previous version I would create a /16 & then under the /16 create a /24. 

Under prefixes the child prefix doesn't nest & when I try to access the child prefixs it throw up the server error.

Eg 
Prefix: 
1.1.0.0/16 
Child Prefix :  
1.1.1.0/24
1.1.2.0/24

If that makes sense.

For what it's worth I have a test version running 3.0.0 I also get an server error which  appears to be the same as #7072

![image](https://user-images.githubusercontent.com/84839725/131834331-515d0896-37b6-4fd1-a23b-0f49bdaab82d.png)

**Reply by Bloodiko** [reply]
I've just tried it on demo.netbox.dev but i cant replicate the issue. 
Weird issue though - no idea. 
Maybe something on the Upgrade broke. 

Do you have the Logs from the Upgrade Process ? - maybe theres a hint on what happened.

**Reply by aquelf** [reply]
Do you mean logs during the upgrade? I've attached the output but don't see any log of any significance.

[3.0.1 upgrade logs.txt](https://github.com/netbox-community/netbox/files/7098472/3.0.1.upgrade.logs.txt)

Thanks

**Reply by aquelf** [reply]
Just looking through Jeremy's response listed in the  #7072  & followed the steps to clear the preferences under the Admin settings & it looks to have fixed the issue, thanks a million for your help.

**Reply by candlerb** [reply]
> What I have noticed that under Sites > Prefixes > Child Prefixs I get the following Server Error?

When that error message is displayed, can you show the *exact* path which is shown in the browser URL bar at that time (with the hostname stripped off)?

Also, can you show the full backtrace?  This is most easily obtained by setting EMAIL and ADMINS in configuration.py, then it will be mailed to you.

I saw [exactly the same problem](https://groups.google.com/g/netbox-discuss/c/5HOBKW3Md30/m/TsiViJtwAgAJ) once at demo.netbox.dev.  But as it's wiped daily I can't reproduce it :-(

"Reverse for 'prefix' with arguments '(None,)'" suggests something weird going on, where it's trying to show a prefix with id "None".

**Reply by aquelf** [reply]
https://x.x.x.x/ipam/prefixes/412/prefixes/

It has got to do with the preferences, as I reset the preferences again & I notices the issue once more, cleared the preferences & it is fine again.

Let me try get the full backtrace

**Reply by candlerb** [reply]
Can you give the exact steps to create and clear the problem?

**Reply by aquelf** [reply]
Upgraded from 2.11.7 to 3.0.1: 
When I Site Groups > Site > Prefixes > choose a prefix > Child Prefix
![image](https://user-images.githubusercontent.com/84839725/131858368-017c7abb-67be-45c5-af31-2c8abb86dbff.png)

To clear:
Admin > Profile & Settings > Preferences > Clear Selected

To Reproduce:
Sites> Prefixes > Configure Table  make changes in this table & the error reappears.

The above steps listed under 'to clear' resolves issue

**Reply by candlerb** [reply]
I don't understand "Sites > Prefixes".  Do you mean "IPAM > Prefixes" in the left hand navigation panel, or something else?

**Reply by aquelf** [reply]
Yes, the same thing. On the left hand panel.

**Reply by candlerb** [reply]
OK, thank you, I reproduced this way:

* IPAM > Prefixes
* Configure Table
* Add column "Prefix (Flat)"
* Save
* Navigate to some prefix which has child prefixes
* Click on "child prefixes" tab: in my case`/ipam/prefixes/17/prefixes/`

Unfortunately it's not E-mailing the exception, I need to dig to find out why.

Removing the extra column (via Configure Table) fixes it.

**Reply by candlerb** [reply]
via DEBUG=True:

```
/opt/netbox/netbox/netbox/views/generic.py, line 70, in get
        return render(request, self.get_template_name(), { 
......
/opt/netbox/venv/lib/python3.8/site-packages/django_tables2/columns/base.py, line 147, in get_attrs
        attrs["href"] = self.compose_url(**kwargs) 
/opt/netbox/venv/lib/python3.8/site-packages/django_tables2/columns/base.py, line 119, in compose_url
        return context.get_absolute_url() 
/opt/netbox/netbox/ipam/models/ip.py, line 305, in get_absolute_url
        return reverse('ipam:prefix', args=[self.pk]) 
```

**Reply by aquelf** [reply]
No problem, I don't seem to be getting the e-mail but yes that sounds exactly like the issue.
